_rules/table-header-cell-has-assigned-cells-d0f69e.md

```diff
@@ -27,20 +27,19 @@ acknowledgments:
 
 ## Applicability
 
-This rule applies to any HTML element with the [semantic role][] of [rowheader][] or [columnheader][] for which all of the following is true:
+This rule applies to any HTML element with the [semantic role][] of [rowheader][] or [columnheader][] within a [table][] or [grid][] element, where the [table][] or [grid][] element is [visible][].
```

Comment:
I am not sure that "within" is unambiguous enough.
I understand that this is trying to address Wilco's first comment of https://www.w3.org/2002/09/wbs/93339/ACTTableHeaders/results#xQ8 but I think that "within" can easily be understood as "descendant of" (this is the meaning we use for ["inside a web page"](https://act-rules.github.io/glossary/#web-page-html), for example).

I agree with the initial comment, but I'm afraid we need to be more precise in explaining what it means for a cell to be part of a table. (and this is somewhat more complex due to the use of roles, so we cannot fully rely on HTML definitions from https://html.spec.whatwg.org/multipage/tables.html#table-processing-model such as "the element corresponds to a cell [=concept, not `td`/`th`] in the table [=concept, not the `table` element]" 😖

Reply:
Something in the lines of "within an element with the semantic role of `table` or `grid` that is its closest ancestor in the accessibility tree"- the ancestor in the accessibility tree does not seem correct because it's used in the DOM standard so have a feeling it should not be used for accessibility tree, but please correct me on that.

Reply:
The problem with "closest ancestor" is that it is the current version and not really working…

Why I understand is that in
```
<table>
<div><div><tr><th>Foo</th></tr></tr>
</table>
```
the `th` is not really "part" of the `table` because the `div` are skipped. This actually gets more complex due to the fact that HTML parsing actually removes the `div` when building the DOM in that case… But we can reach a similar result with roles:
```html
<div role="table">
<div><div>
  <div role="tablerow"><div role="columnheader">Foo</div></div>
</div></div>
</div>
```
But then in that case it's not even clear what "being part of the table" means since the algorithm for forming a table only works on HTML tables, not on ARIA ones…

Using "within" is, I think, not good because of this:
```html
<table>
<tr><td>
  <table hidden>
  <tr><th>Foo</th><tr>
  </table>
</td></tr>
</table>
```
In which case, the `th` is arguably "within" both `table` elements (as in "within the subtree rooted at that element"), but we want to exclude it from the rule, so we need a definition of "within a table"… And copying the table building algorithm doesn't seem smart…